### DISCUSSION TITLE
Automatic update of the Insiders Docker image

### DISCUSSION THREAD

**Discussion by white-gorilla** [squidfunk/mkdocs-material#4874] [Q&A] [answered]
Hello,

Tell me if this is not the place to post this kind of question.

Last Thursday, we chose to become a sponsor to support your excellent work.
I took the liberty of testing the insiders version.

To do this, I followed the installation guide to generate a docker image in our private Github Packages registry. Everything went well, I manually released the current version `4.27.1` (with this tag) and the image was successfully pushed to the registry (which allowed me to do my tests).

I saw this morning that an [update `4.28.0`](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.28.0) was made this weekend. Having installed and configured `Pull App` on the fork. I expected the repo. to be synchronized, a PR to be created with new tag and a new image to be created in the registry.
This did not happen.

As long as an image was generated manually, and Pull App is installed in the fork, I don't see what I missed.

Does this seem normal to you?

---
I was wondering if it was not a problem of rights. The Sync Fork button indicates:
```
This branch has conflicts that must be resolved

Discard 0 commits to make this branch match the upstream repository. 0 commits will be removed from this branch.

You can resolve merge conflicts using the command line and a text editor.
```

And I can't sync manually:`You can't read the parent repository.  `

**Comment by squidfunk** [comment]
As far as I know, the Pull Request App won't create releases, so you would need to create a release for each tag. It might be possible to create (or use) a GitHub action for that, which will be triggered on new tags and create releases automatically. If you cook something up, please share it here. There was a prior discussion somewhere here, but I can't find it right now.

**Reply by squidfunk** [reply]
Additionally, on your other note:

```
This branch has conflicts that must be resolved

Discard 0 commits to make this branch match the upstream repository. 0 commits will be removed from this branch.

You can resolve merge conflicts using the command line and a text editor.
```

I've never seen this and it sounds weird – according to the message, there shouldn't be any problems, but still the sync button doesn't work. What does the GitHub Pull Request App say? Are you trying to sync the fork with an account that has access to Insiders?

**Reply by white-gorilla** [reply]
> As far as I know, the Pull Request App won't create releases, so you would need to create a release for each tag. It might be possible to create (or use) a GitHub action for that, which will be triggered on new tags and create releases automatically. If you cook something up, please share it here. There was a prior discussion somewhere here, but I can't find it right now.

Thank you very much, I must have misunderstood.
I thought that the Pull Request App creates a PR, with the changes, and adds a new tag. And that this new tag is picked up by the `publish.yaml` workflow. And that it automatically builds and publishes the Docker image to my private registry.
What you are telling me is that for the moment, by default, the PR and the tag are created automatically, but that I have to create the release myself each time?

For the rest, unless I'm mistaken, the repo. doesn't seem to be automatically synchronized by the Pull Request App. I forgot to mention that it was forked in our orga. by the account that has the rights.  The only thing I did was to manually trigger with `https://pull.git.ci/process/${owner}/${repo}`, which returns `success`.

But if I look at the latest commits, they are from January 12 and the Insiders version is still `4.27.1` :
```
mkdocs-material-9.0.4 (2022-01-12)

  * Fixed #4823: Improved contrast ratio in footer (9.0.2 regression)
  * Fixed #4832: Set navigation items back to black (9.0.3 regression)
  * Fixed #4843: Emojis broken due to maxcdn.com shutting down
  * Upgraded Python Markdown Extensions to 9.9.1

mkdocs-material-9.0.3+insiders-4.27.1 (2022-01-08)
```
I would ask the person who has the rights tomorrow to do the test.

**Comment by white-gorilla** [comment, marked as answer]
Hello,

I checked with the person who has the rights (we'll call him Michael for simplicity). He doesn't see the same thing as me on the fork sync button. It just indicates that the repo is not up to date, that he can compare or synchronize. He has the possibility to synchronize (but not me. I'm a fork admin).

Finally, isn't it the same problem with Pull App? I have the impression that only the person/account who has the rights can act.

Note: I will manually create a new release with the tag `4.28.0`.

**Reply by squidfunk** [reply]
It is definitely not related to Insiders, but a problem with how the syncing is done. I'm not sure whether it would be possible for Pull App to create the release, maybe ask the maintainers?

**Reply by white-gorilla** [reply]
OK, thanks, before I close the topic, I feel like it's the fork in the organization that's the problem, do you think if I create a dedicated account and fork into that account it might be better? That's one of the recommendations, to have a dedicated bot account. I'll plan the migration.

**Reply by squidfunk** [reply]
I don't think that'll solve the problem of synchronizing. We're actively working on a solution with which the forking shouldn't be necessary anymore (= organizations can manage collaborators for themselves). However, we need to think about how to handle the Docker image then. Let me do some thinking and come back to you 😊

**Reply by white-gorilla** [reply]
OK. Thanks @squidfunk.

**Reply by purpleclay** [reply]
It would be great to have the ability to keep our insiders fork in sync. I was puzzled by this also, as the documentation states that installing Pull App will keep our fork in sync with the upstream. I don't see that behaviour and have to manually pull when I remember. If you come up with a solution that doesn't require the forking that would be great.

**Reply by squidfunk** [reply]
We're already working on a solution for this problem. In particular, sponsors will be able to manage collaborators themselves, which will make forking unnecessary. Additionally, the Insiders repository will be moved to an organization.

All of this will be available later this year.
